### ISSUE TITLE
tensorflow-io takes forever to build in linux-sandbox ubuntu

### ISSUE THREAD

**Issue by marioecd** [tensorflow/io#926]
I've tried to do bazel build //tensorflow_io/... in a ubuntu machine with a 100GB disk. It takes forever to compile. It seems to use a linux-sandbox when I do bazel build there. The OS version is:

NAME="Ubuntu"
VERSION="18.04.4 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.4 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic

However, on a Debian OS machine, the bazel build takes only about 20 minutes. Bazel seems to run a processwrapper-sandbox there.

Is there any particular bazel setting in configure.sh (or else) that could make compilation take so long (many hours) in ubuntu OS?

**Comment by yongtang** [comment]
Bazel is not exactly memory/cpu efficient, but it normally takes about 20 min or so on a decent machine to build tensorflow/io. Maybe you can try:
```
bazel build -s --verbose_failurees //tensorflow_io/... 
```

to see where it spend most of the time.

Things I encounter that may take very long in the past include the network. If bazel has not download the third_party packages before, then bazel will download first and this could varies.

I noticed in one instance somehow bazel was downloading one package close to my internet bandwidth but the next min it downloading another package in a much slower 30Kb/s. In those situations Ctrl-C then restart the build typically resolves the issue.

**Comment by marioecd** [comment]
Yeah I've been running it with **bazel build -s --verbose_failurees //tensorflow_io/...** 
I see it taking a long time on everything and anything. Gets stuck compiling any subcomponent that would have taken 20 seconds to build on Debian. The interesting thing is that in Debian it is always 20 minutes fast, while on the ubuntu os it is always hours slow. Both machines, though different type of machines, run in GCP and from the same zone. That's why I get the sense that this may be a OS setting, since that seems to be the most significant difference.

**Comment by yongtang** [comment]
@marioecd Wondering what is the memory size of your machine? The bazel itself is written in Java and is really a resource intensive process. (could be several GBs of memory just for bazel, not counting gcc).

Our GitHub CI is testing against Ubuntu 18.04 and 20.04 by default:
https://github.com/tensorflow/io/runs/626946045?check_suite_focus=true

(GitHub CI is using Azure's VM. I don't think there is fundamental difference between Azure and GCP.)

On GitHub CI it is around 30-40min+ or so but not in hours. Also some of the GitHub CI configurations uses optimization=O2 which takes longer. So in theory by default (no additional flag passed to `bazel build`) it takes around 20-30min or so I would assume.

**Comment by marioecd** [comment]
Ah good point. Just checked. It's not memory size though. It has a lot of memory.
